controllers/storagecluster/external_resources.go

```diff
@@ -74,6 +74,9 @@ func (r *StorageClusterReconciler) setRookCSICephFS(
 	if rookCephOperatorConfig.Data[rookEnableCephFSCSIKey] == enableDisableFlagStr {
 		return nil
 	}
+	if rookCephOperatorConfig.Data == nil {
+		rookCephOperatorConfig.Data = map[string]string{}
+	}
```

Comment:
I think you mislocated the check - line 74 is already trying to access the map.

Reply:
It should work on line 77 as well because on line 74 it is a comparison and not an assignment but I have updated in the latest commit

Reply:
checking nil map works the map should not be empty when you are trying to assign it.

Reply:
Just checked and you guys are right.
But to me, it seems like a quirk of the language.